app/internal/page/auth_redirect.go

```diff
@@ -63,6 +66,24 @@ func AuthRedirect(logger Logger, c authRedirectClient, store sessions.Store, sec
 			return
 		}
 
-		http.Redirect(w, r, paths.YourDetails, http.StatusFound)
+		stateJson, err := base64.StdEncoding.DecodeString(r.FormValue("state"))
+		if err != nil {
+			logger.Print(fmt.Sprintf("Error base64 decoding state: %s", err))
+			return
+		}
+
+		var state State
+		if err = json.Unmarshal(stateJson, &state); err != nil {
+			logger.Print(fmt.Sprintf("Error unmarshalling state JSON: %s", err))
+			return
+		}
+
+		redirectPath := paths.YourDetails
+
+		if state.Locale == "cy" {
```

Comment:
or when I said encode in the state param as an option I was thinking something more like prefixing the existing value with `cy-` if Welsh rather than make it JSON

Reply:
Yeah, this would be much less complex. I completely glossed over the cookie side of things. I'll move the locale to the cookie instead.